lib/bson/regexp.rb

```diff
@@ -161,8 +157,8 @@ def initialize(pattern, options = '')
           if options.to_s.include?(NULL_BYTE)
             raise Error::InvalidRegexpPattern, "Regexp options cannot contain a null byte: #{options}"
           end
-        elsif !options.is_a?(Integer)
-          raise ArgumentError, "Regexp options must be a String, Symbol, or Integer"
+        else
+          raise ArgumentError, "Regexp options must be a String or Symbol"
```

Comment:
Do you think we should disallow symbols for options? I can make a follow-up ticket if so.

Reply:
I personally don't have a problem keeping symbol options... If you have a specific reason then I wouldn't mind dropping it but otherwise we can keep it.

Reply:
:+1: That's fine.